src/main/java/com/epam/machine/controller/CertainOrderServlet.java

```diff
@@ -37,4 +39,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) {
             logger.error(e.getMessage());
         }
     }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) {
+        int statusId = Integer.parseInt(request.getParameter("status"));
+        String status = offers.get(offerId).getStatus();
```

Comment:
and here `offerService.get(offerId)` ?

Reply:
Same here, bug was in adminorders.jsp, fixed it